presto-main/src/main/java/com/facebook/presto/operator/LayoutSpecificAriaHash.java

```diff
@@ -268,4 +270,131 @@ private void insertHashes(long[] hashes, long[] entries, int fill)
             while (true);
         }
     }
+
+    public static class AriaLookupSource
+            implements LookupSource
+    {
+        int statusMask;
+        long[] status;
+        long[] table;
+        Slice[] slices = new Slice[16];
+
+        private int[] fill = new int[16];
+        private int currentSlab = -1;
+
+        private long allocBytes(int bytes)
+        {
+            if (currentSlab == -1 || fill[currentSlab] + bytes > (128 * 1024)) {
+                long w = newSlab();
+                fill[currentSlab] = bytes;
+                return w;
+            }
+            int off = fill[currentSlab];
+            fill[currentSlab] += bytes;
+            return (((currentSlab) << 17) + (off));
+        }
+
+        private long newSlab()
+        {
+            ++currentSlab;
+            if (slices.length <= currentSlab) {
+                int newSize = slices.length * 2;
+                slices = Arrays.copyOf(slices, newSize);
+                fill = Arrays.copyOf(fill, newSize);
+            }
+            Slice s = AriaHash.getSlice();
+            slices[currentSlab] = s;
+            return (currentSlab) << 17;
+        }
+
+        private void release()
+        {
+            for (Slice slice : slices) {
+                if (slice != null) {
+                    AriaHash.releaseSlice(slice);
+                }
+            }
+        }
+
+        private void setSize(int count)
+        {
+            int size = 8;
```

Comment:
Extract these to static constants, will help with understanding what or why `1.3` was chosen for example (i.e. with the name of the constant).